src/Particular.ThroughputCollector.Persistence.RavenDb/ThroughputDataStore.cs

```diff
@@ -1,22 +1,131 @@
+
+        var documentId = id.GenerateDocumentId();
+
+        var document = await session.LoadAsync<EndpointDocument>(
+            documentId,
+            builder => builder.IncludeTimeSeries(ThroughputTimeSeriesName),
+            cancellationToken);
+
+        var endpoint = document?.ToEndpoint();
+        if (endpoint != null)
+        {
+            var timeSeries = await session
+                .IncrementalTimeSeriesFor(documentId, ThroughputTimeSeriesName)
+                .GetAsync(token: cancellationToken);
+
+            endpoint.LastCollectedDate = DateOnly.FromDateTime(timeSeries.LastOrDefault()?.Timestamp ?? DateTime.MinValue);
+        }
+
+        return endpoint;
+    }
+
+    public async Task<IEnumerable<(EndpointIdentifier, Endpoint?)>> GetEndpoints(IEnumerable<EndpointIdentifier> endpointIds, CancellationToken cancellationToken = default)
+    {
+        var endpoints = new List<(EndpointIdentifier, Endpoint?)>();
+
+        using var session = store.OpenAsyncSession(databaseConfiguration.Name);
+
+        var documentIdLookup = endpointIds.ToDictionary(
+            endpointId => endpointId,
+            endpointId => endpointId.GenerateDocumentId());
+
+        var endpointDocuments = await session.LoadAsync<EndpointDocument>(
+            documentIdLookup.Values.Distinct(),
+            builder => builder.IncludeTimeSeries(ThroughputTimeSeriesName),
+            cancellationToken);
+
+        foreach (var (documentId, endpointDocument) in endpointDocuments)
+        {
+            var id = endpointDocument == null
+             ? endpointIds.First(id => id.GenerateDocumentId().Equals(documentId))
+             : endpointDocument.EndpointId;
+
+            var endpoint = endpointDocument?.ToEndpoint();
+            if (endpoint != null)
+            {
+                var timeSeries = await session
+                    .IncrementalTimeSeriesFor(documentId, ThroughputTimeSeriesName)
+                    .GetAsync(token: cancellationToken);
+
+                endpoint.LastCollectedDate = DateOnly.FromDateTime(timeSeries.LastOrDefault()?.Timestamp ?? DateTime.MinValue);
+            }
+
+            endpoints.Add((id, endpoint));
+        }
+
+        return endpoints;
+    }
+
+    public async Task SaveEndpoint(Endpoint endpoint, CancellationToken cancellationToken = default)
+    {
+        var document = endpoint.ToEndpointDocument();
+
+        using var session = store.OpenAsyncSession("throughput");
+
+        await session.StoreAsync(document, document.GenerateDocumentId(), cancellationToken);
+        await session.SaveChangesAsync(cancellationToken);
+    }
+
+    public async Task RecordEndpointThroughput(EndpointIdentifier id, IEnumerable<EndpointThroughput> throughput, CancellationToken cancellationToken = default)
+    {
+        if (!throughput.Any())
+        {
+            return;
+        }
+
+        using var session = store.OpenAsyncSession("throughput");
```

Comment:
So I assume we can name the sessions, for some kind of telemetry purposes?
Should the session name include more context, eg "Throughput/RecordEndpointThroughput"

Reply:
Sorry, so what is "throughput" argument passed?

Reply:
It specifies which database to use.  The default database in the document store is the SC one.